src/Datasource/EntityTrait.php

```diff
@@ -482,7 +482,7 @@ public function setVirtual(array $properties, $merge = false)
             return $this;
         }
 
-        $this->_virtual += $properties;
+        $this->_virtual = array_merge($this->_virtual, $properties);
```

Comment:
What happens when the same property is set more than once? I don't think we want duplicates.